File: app/src/main/scala/com/trade/enrichedService/EnrichmentService.java
package com.trade.enrichedService;

import com.crankuptheamps.client.HAClient;
import com.crankuptheamps.client.Message;
import com.crankuptheamps.client.MessageStream;
import com.trade.amps.AmpsClientUtil;

public class EnrichmentService {
    public static void main(String[] args) throws Exception {
        HAClient client = AmpsClientUtil.connect("EnrichmentService");
        System.out.println("Enrichment Service connected, waiting for trades...");

        MessageStream ms = client.subscribe("trades.raw").timeout(0);

        int count = 0; // debug limit, remove/comment in production
        for (Message msg : ms) {
            String trade = msg.getData();
            String enrichedTrade = trade.replace("}", ", \"enriched\":true}");
            client.publish("trades.enriched", enrichedTrade);
            System.out.println("Enriched and Published trade: " + enrichedTrade);

            if (++count >= 5) break; // remove in production
        }

        client.close();
    }
}

File: app/src/main/scala/com/trade/enrichedService/EnrichmentService.scala
//package com.trade.enrichedService
//
//import com.crankuptheamps.client.{HAClient, Message}
//import com.trade.amps.AmpsClientUtil
//import com.trade.common.Trade
//import play.api.libs.json.Json
//
//object EnrichmentService extends App {
//
//  val client: HAClient = AmpsClientUtil.connect("EnrichmentService")
//  println("Enrichment Service connected")
//
//  val ms = client.subscribe("trades.raw").timeout(0)
//
//  while(true) {
//    val msg = ms.next()
//    val trade = Json.parse(msg.getData).as[Trade]
//
//    val enrichedTrade = trade.copy(enriched = true)
//    val json = Json.stringify(Json.toJson(enrichedTrade))
//
//    client.publish("trades.enriched", json)
//    println(s"Enriched trade: $json")
//  }
//}
